diff --git a/services/licensemanager/LicenseManager.go b/services/licensemanager/LicenseManager.go
index 56fd9f0..f625bcf 100644
--- a/services/licensemanager/LicenseManager.go
+++ b/services/licensemanager/LicenseManager.go
@@ -265,7 +265,7 @@ func (lm *LicenseManager) disableService(service *Service) {
 	}
 
 	// Set settings
-	if _, err = settings.SetSettings(settingsSegs, newSettings, true); err != nil {
+	if _, err = settings.SetSettings(settingsSegs, newSettings, true, false); err != nil {
 		lm.logger.Warn("Failed to set disabled settings for service %s\n", service.Name)
 	}
 }
diff --git a/services/settings/settings.go b/services/settings/settings.go
index 3b42d77..6e74fb9 100644
--- a/services/settings/settings.go
+++ b/services/settings/settings.go
@@ -85,8 +85,8 @@ func GetSettings(segments []string) (interface{}, error) {
 }
 
 // SetSettings updates the settings
-func SetSettings(segments []string, value interface{}, force bool) (interface{}, error) {
-	return SetSettingsFile(segments, value, settingsFile, force)
+func SetSettings(segments []string, value interface{}, force bool, skip bool) (interface{}, error) {
+	return SetSettingsFile(segments, value, settingsFile, force, skip)
 }
 
 // TrimSettings trims the settings
@@ -136,7 +136,7 @@ func GetSettingsFile(segments []string, filename string) (interface{}, error) {
 }
 
 // SetSettingsFile updates the settings
-func SetSettingsFile(segments []string, value interface{}, filename string, force bool) (interface{}, error) {
+func SetSettingsFile(segments []string, value interface{}, filename string, force bool, skip bool) (interface{}, error) {
 	var ok bool
 	var err error
 	var jsonSettings map[string]interface{}
@@ -158,7 +158,7 @@ func SetSettingsFile(segments []string, value interface{}, filename string, forc
 	}
 
 	saveLocker.Lock()
-	output, err := syncAndSave(jsonSettings, filename, force)
+	output, err := syncAndSave(jsonSettings, filename, force, skip)
 	saveLocker.Unlock()
 	if err != nil {
 		var errJSON map[string]interface{}
@@ -356,7 +356,7 @@ func TrimSettingsFile(segments []string, filename string) (interface{}, error) {
 		}
 	}
 
-	output, err := syncAndSave(jsonSettings, filename, false)
+	output, err := syncAndSave(jsonSettings, filename, false, false)
 	if err != nil {
 		return map[string]interface{}{"error": err.Error(), "output": output}, err
 	}
@@ -416,8 +416,8 @@ func getSettingsFromJSON(jsonObject interface{}, segments []string) (interface{}
 }
 
 // runSyncSettings runs sync-settings on the specified filename
-func runSyncSettings(filename string, force bool) (string, error) {
-	cmd := exec.Command("/usr/bin/sync-settings", "-f", filename, "-v", "force="+strconv.FormatBool(force))
+func runSyncSettings(filename string, force bool, skip bool) (string, error) {
+	cmd := exec.Command("/usr/bin/sync-settings", "-f", filename, "-v", "force="+strconv.FormatBool(force), "-w", "skip:="+strconv.FormatBool(skip))
 	outBytes, err := cmd.CombinedOutput()
 	jsonOutput, data, errParse := parseSyncSettingsJsonOutput(outBytes)
 
@@ -473,7 +473,7 @@ func syncSystemFiles() {
 // it copies the tmp file to the destination specified in filename
 // if sync-settings does not succeed it returns the error and output
 // returns stdout, stderr, and an error
-func syncAndSave(jsonObject map[string]interface{}, filename string, force bool) (string, error) {
+func syncAndSave(jsonObject map[string]interface{}, filename string, force bool, skip bool) (string, error) {
 	// we want this to run after all files have been closed
 	// so we defer it as soon as possible
 	defer syncSystemFiles()
@@ -492,7 +492,7 @@ func syncAndSave(jsonObject map[string]interface{}, filename string, force bool)
 		return "Failed to write settings.", err
 	}
 
-	output, err := runSyncSettings(tmpfile.Name(), force)
+	output, err := runSyncSettings(tmpfile.Name(), force, skip)
 	scanner := bufio.NewScanner(strings.NewReader(output))
 	for scanner.Scan() {
 		logger.Info("sync-settings: %v\n", scanner.Text())
diff --git a/services/settings/settings_file.go b/services/settings/settings_file.go
index 9fb9872..9443b7e 100644
--- a/services/settings/settings_file.go
+++ b/services/settings/settings_file.go
@@ -132,7 +132,7 @@ func (file *SettingsFile) SetSettingsNoSync(segments []string, value any) error
 }
 
 // SetSettings updates the settings. Calls lock/unlock on the SettingsFile's mutex
-func (file *SettingsFile) SetSettings(segments []string, value interface{}, force bool) (interface{}, error) {
+func (file *SettingsFile) SetSettings(segments []string, value interface{}, force bool, skip bool) (interface{}, error) {
 	var ok bool
 	var err error
 	var jsonSettings map[string]interface{}
@@ -154,7 +154,7 @@ func (file *SettingsFile) SetSettings(segments []string, value interface{}, forc
 	}
 
 	file.mutex.Lock()
-	output, err := syncAndSave(jsonSettings, file.filename, force)
+	output, err := syncAndSave(jsonSettings, file.filename, force, skip)
 	file.mutex.Unlock()
 	if err != nil {
 		var errJSON map[string]interface{}
@@ -232,7 +232,7 @@ func (file *SettingsFile) SetAllSettingsWithExceptions(newSettings map[string]in
 	newSettings["system"].(map[string]interface{})["httpPort"] = currentSettings["system"].(map[string]interface{})["httpPort"].(string)
 	newSettings["system"].(map[string]interface{})["httpsPort"] = currentSettings["system"].(map[string]interface{})["httpsPort"].(string)
 
-	return file.SetSettings(nil, newSettings, true)
+	return file.SetSettings(nil, newSettings, true, false)
 }
 
 // Generates a backup of a settings file using a provided script. Locks the settings file before generation.
